influxdb/src/client/mod.rs

```diff
@@ -301,12 +301,15 @@ impl Client {
 
 pub(crate) fn check_status(res: &HttpResponse) -> Result<(), Error> {
     let status = res.status();
-    if status == StatusCode::UNAUTHORIZED.as_u16() {
-        Err(Error::AuthorizationError)
-    } else if status == StatusCode::FORBIDDEN.as_u16() {
-        Err(Error::AuthenticationError)
-    } else {
-        Ok(())
+    match status {
+        StatusCode::OK => Ok(()),
+        code => {
+            if code.as_str().as_bytes()[0] == b'2' {
```

Comment:
Possibly stupid question: Why are we string matching instead of calling [`is_success()`](https://docs.rs/http/0.2.12/http/status/struct.StatusCode.html#method.is_success)?

Reply:
> Yes of course, my misunderstanding, I thought we dont wanna use any StatusCode methods.

Reply:
Well, `.as_str()` is another status code method, and using those methods is perfectly fine so long they are independent of the http backend (which they appear to be).